thorlcr/activities/thdiskbaseslave.cpp

```diff
@@ -671,6 +671,8 @@ void CDiskWriteSlaveActivityBase::processDone(MemoryBuffer &mb)
     modifiedTime.getTime(hour, min, sec, nanosec);
     modifiedTime.setTime(hour, min, sec, 0);
     modifiedTime.serialize(mb);
+    if (diskHelperBase->getFlags() & TDXtemporary && !container.queryJob().queryUseCheckpoints())
+        container.queryTempHandler()->setSize(fName, sz);
```

Comment:
I was looking at modifying registerFile to return CFileUsageEntry.   CMasterGraphTempHandler::registerFile  doesn't always create a CFileUsageEntry.  Should CMasterGraphTempHandler::registerFile return a nullptr if calls wu->addFile() rather than CGraphTempHandler::registerFile?

Reply:
yes, it will only create/add a CFileUsageEntry if it's not a published file.
I'd change it to return null where needed.